cachito/web/api_v1.py

```diff
@@ -28,15 +30,23 @@ def get_requests():
         value exceeds configuration's CACHITO_MAX_PER_PAGE.
     :rtype: flask.Response
     """
+    # Check if user is filtering requests by state
+    state = flask.request.args.get('state', default='')
     # Default versbose flag to False
     verbose = str_to_bool(flask.request.args.get('verbose', False))
     max_per_page = flask.current_app.config['CACHITO_MAX_PER_PAGE']
     # The call to `paginate` will inspect the current HTTP request for the
     # pagination parameters `page` and `per_page`.
     pagination_query = Request.query.paginate(max_per_page=max_per_page)
     requests = pagination_query.items
+    if state != '':
+        state_int = RequestStateMapping.__members__[state].value
+        items = [request.to_json(verbose=verbose) for request in requests
```

Comment:
You are filtering at the Python level and after pagination occurs. Instead, you need to add a SQLAlchemy filter to `Request.query` before `.paginate(max_per_page=max_per_page)` is called on it. This will filter at the SQL level which is much more efficient and will cause the pagination to be done on the filtered results.